manager/controlapi/secret.go

```diff
@@ -12,13 +17,40 @@ import (
 // MaxSecretSize is the maximum byte length of the `Secret.Spec.Data` field.
 const MaxSecretSize = 500 * 1024 // 500KB
 
+var isValidSecretName = regexp.MustCompile(`^[a-zA-Z0-9](?:[a-zA-Z0-9-_.]{0,62}[a-zA-Z0-9])*$`)
```

Comment:
IRL discussion:  We just want to make sure the names would work for what we're going to use it for (for example, in case we wanted to be able to use the name for environment variables).  Which I think excludes `-` and `.`  However, if we wanted them to also work for filenames maybe having a `.` is ok?  

Either way, I think we were going with requiring the user give us a `target` value, and it may make sense to let users be a little more expressive in their names (to make it easier to script naming conventions, etc.) if they need to be able to describe what the secret is.

I don't feel super strongly - I'll leave this as is for now, unless someone has a stronger opinion (and we can also resolve this in a later PR).  I have made the variable name change, and updated the length checking to happen in the validation function instead of in the regex.